docs/rfcs/0007-deployment-flow.md

```diff
@@ -0,0 +1,111 @@
+
+![image](assets/deployment-flow-region.png)
+
+All the above requirements share the same thing in the context, that is: it can be done by users' deployment for their applications as PipeCD applications one by one and make trigger those deployments manually via console or make trigger via pull requests to those application configurations separately, but this all manual stub are tedious and difficult to manage smoothly. With this new PipeCD deployment flow feature, all of those manual steps will be replaced, to keep the good point of using a CD system.
+
+# Detailed design
+The idea is to keep the PipeCD application deployment as a unit of deployment as is, but add __a way to enable users to manipulate a deployment flow based on the state of the last sucessfully run deployment__.
+
+A canonical flow would look as below:
+- Users trigger to run their first application (the first application in their deployment flow) via the web console or pull requests as usual (1)
+- Deploy the first triggered application in the deployment flow (2)
+- After the application is deployed successfully, the piped will send a trigger command to trigger the deployment flow to the control-plane, since all those applications are shared in the same project, this trigger should be valid (3)
+- Control-plane gets deployment events triggered by the first application's piped and makes sync deployment commands for all listed applications (4)
+- Pipeds which handle deployment for all triggered applications fetch sync command and deploy its application once the conditions are satisfied (5)
+
+![image](assets/deployment-flow-idea.png)
+
+## Configuration
+
+At the step (3), to enable configure the deployment flow after the first applications (in the deployment flow) deployed successfully, we introduce a new configuration section to the deployment configuration named `postSync` which contains configurations of actions that the in charge piped should do after the deployment run successfully.
+
+The configuration would look like:
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  input:
+    ...
+  pipeline:
+    ...
+  trigger:
+    paths:  
+    onEvent:
+    onChange:
+    onFlow:
+      disable: false
+  postSync:
+    flow:
+      graph:
+        - appName: lambda1
+          requireApproval: true # add wait approval stage to the lambda 1 deployment pipeline on planning 
+        - appName: lambda2
+          depends:
+            - lambda1
+      conditions:
+        - commitPrefix: “xx”
+        - ...
+```
+
+The newly added configuration fields as follow:
+
+### postSync.flow
+
+| Field | Type | Description | Required |
+|-|-|-|-|
+| graph | []GraphNode | The list of applications which should be triggered once deployment of this application rolled out successfully. | Yes |
```

Comment:
Possibly you are aware of it, but currently the application name isn't unique within a project so additional something is needed.